siriuspy/siriuspy/thread.py

```diff
@@ -114,3 +115,95 @@ def stop(self):
         """Stop execution."""
         self._unpaused.set()
         self._stopped.set()
+
+
+class DequeThread(_deque):
+    """DequeThread.
+
+    This class manages generic operations (actions) using an append-right,
+    pop-left queue. Each operation processing is a method invoked as a separate
+    thread.
+    """
+
+    _lock = _Lock()
+
+    def __init__(self):
+        """Init."""
+        self._thread = None
+        self._ignore = False
+        self._enabled = True
+
+    @property
+    def last_operation(self):
+        """Return last operation."""
+        return self._last_operation
+
+    @property
+    def enabled(self):
+        """Enable process."""
+        return self._enabled
+
+    @enabled.setter
+    def enabled(self, value):
+        self._bsmpcomm = value
+
+    def ignore_set(self):
+        """Turn ignore state on."""
+        self._ignore = True
+
+    def ignore_clear(self):
+        """Turn ignore state on."""
+        self._ignore = False
+
+    def append(self, operation, unique=False):
+        """Append operation to queue."""
+        DequeThread._lock.acquire(blocking=True)
+        if not self._ignore:
+            if not unique:
+                super().append(operation)
+                self._last_operation = operation
+            else:
+                # super().append(operation)
```

Comment:
I think it is not necessary to keep these comment lines.

Reply:
done.